Nodejs/Product/Npm/SPI/NpmInstallCommand.cs

```diff
@@ -2,15 +2,20 @@
 
 namespace Microsoft.NodejsTools.Npm.SPI
 {
-    internal class NpmInstallCommand : NpmCommand
+    public class NpmInstallCommand : NpmCommand
     {
+        public static bool IsInstalling { get; private set; } = false;
+
         public NpmInstallCommand(
             string fullPathToRootPackageDirectory,
             string pathToNpm = null,
             bool useFallbackIfNpmNotFound = true)
             : base(fullPathToRootPackageDirectory, showConsole: false, pathToNpm: pathToNpm)
         {
             this.Arguments = "install";
+
+            this.CommandStarted += (sender, eventArgs) => { IsInstalling = true; };
+            this.CommandStarted += (sender, eventArgs) => { IsInstalling = false; };
```

Comment:
>CommandStarted [](start = 17, length = 14)

Was this supposed to be `CommandStopped`?

Reply:
Good, catch those where last minute changes and I didn't noticed that I have used the incorrect handler. 
Also the code that executed the event handlers didn't handle exceptions correctly. I have refactored that code as well. Check `NpmCommand.cs` file.